services/src/main/java/org/keycloak/protocol/AuthorizationEndpointBase.java

```diff
@@ -188,16 +189,20 @@ protected AuthenticationSessionModel createAuthenticationSession(ClientModel cli
             UserSessionModel userSession = userSessionCrossDCManager.getUserSessionIfExistsRemotely(manager, realm);
 
             if (userSession != null) {
-                String userSessionId = userSession.getId();
-                rootAuthSession = session.authenticationSessions().createRootAuthenticationSession(userSessionId, realm);
-                authSession = rootAuthSession.createAuthenticationSession(client);
-                logger.debugf("Sent request to authz endpoint. We don't have root authentication session with ID '%s' but we have userSession." +
-                        "Re-created root authentication session with same ID. Client is: %s . New authentication session tab ID: %s", userSessionId, client.getClientId(), authSession.getTabId());
+                UserModel user = userSession.getUser();
+                if (user != null && !user.isEnabled()) {
+                    authSession = createNewAuthenticationSession(manager, client);
+
+                    session.sessions().removeUserSession(realm, userSession);
```

Comment:
@martin-kanis Based on the discussion on keycloak-dev, we will handle the case of disabled user differently than the case of user registration. So I am sorry for the confusion :)

Still I believe it will be slightly better to call "authenticationManager.backchannelLogout" instead of directly call just "removeUserSession" . That would mean an attempt to propagate to logout to the applications or identity brokers, which is good. Is it possible? I will accept this PR afterwards.

Reply:
Done